### DISCUSSION TITLE
Assert that collection contains at least one element satisfying another assertion

### DISCUSSION THREAD

**Discussion by guythetechie** [fluentassertions/fluentassertions#2043] [Ideas]
Let's say I have a collection:
```csharp
var names = new[] { "John", "James" };
```

I can validate all names have length 4 by:
```csharp
names.Should().AllSatisfy(name => name.Should().HaveLength(4));
```

To validate that **at least one** name satisfies the condition, I'm currently doing:
```csharp
names.Should().Contain(name => name.Length == 4);
```

Is there a way to do this and keep the fluent syntax? Something like:
```csharp
names.Should().Contain(name => name.Should().HaveLength(4));
```

Thanks!

**Comment by dennisdoomen** [comment]
There's no elegant way to do that. The closest you can get is by using `Match`.
